types/mongoose/index.d.ts

```diff
@@ -2435,8 +2435,8 @@ declare module "mongoose" {
      * If a callback is not passed, the aggregate itself is returned.
      * @param aggregations pipeline operator(s) or operator array
      */
-    aggregate(aggregations: any[]): Aggregate<any[]>;
-    aggregate(aggregations: any[], cb: Function): Promise<any[]>;
+    aggregate(aggregations?: any[]): Aggregate<any[]>;
+    aggregate(aggregations: any[], cb?: Function): Promise<any[]>;
```

Comment:
I don't think it makes sense to make `cb` optional here; does passing in `undefined` as a second argument actually return a `Promise`?

Reply:
Yes you don't need to pass a ```callback function``` if you are using ```Promise```.